lib/relooper/src/graph/reduction/dj_graph.rs

```diff
@@ -0,0 +1,97 @@
+use std::{
+    borrow::Borrow,
+    collections::{HashMap, HashSet},
+    hash::Hash,
+};
+
+use crate::graph::{
+    cfg::{Cfg, CfgLabel},
+    domtree::DomTree,
+    GEdgeColl, Graph, GraphMut,
+};
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum DJEdge<T> {
```

Comment:
Please add some doc comments here about what this enum is and what the variants represent.

Reply:
fixed